ansible_base/lib/serializers/common.py

```diff
@@ -98,3 +101,15 @@ class Meta(CommonModelSerializer.Meta):
 class ImmutableCommonModelSerializer(AbstractCommonModelSerializer):
     class Meta(AbstractCommonModelSerializer.Meta):
         fields = AbstractCommonModelSerializer.Meta.fields + ['created', 'created_by']
+
+
+class UneditableSystemUserSerializer(CommonModelSerializer):
+    """
+    Disallows editing of system user (settings.SYSTEM_USERNAME).
+    Make sure to call super if you override validate(..).
+    """
+
+    def validate(self, data):
+        if hasattr(self.instance, "id") and self.instance.id == models.get_system_user().id:
```

Comment:
It should always have an id (or at least a pk), but `hasattr(self, 'instance')` might be a necessary check.

Reply:
I found that id isn't present for creates (since this runs before it gets populated by the ORM).  But I didn't see a case where instance wasn't populated.  No problem to add that though.